docs/references/contributor_guide.md

```diff
@@ -1,22 +1,126 @@
 
-See [Install SGLang, Method 2: From Source section](../start/install.md).
+## 1. Setting Up & Building from Source
 
-## Format Your Code
-Use these commands to format your code and pass CI linting tests.
+1. **Clone the Repository**
 
-```
+bash
+   git clone https://github.com/sgl-project/sglang.git
+   cd sglang
+
+2. **Install Dependencies & Build**
+   Refer to our [Install SGLang](https://sgl-project.github.io/start/install.html) documentation for a detailed explanation of setting up the necessary dependencies and compiling the codebase locally.
+
+## 2. Code Formatting with Pre-Commit
+
+We use [pre-commit](https://pre-commit.com/) to maintain consistent code quality and style checks. Before pushing your changes, please run:
+
+bash
 pip3 install pre-commit
 cd sglang
 pre-commit install
 pre-commit run --all-files
-```
 
-## Add Unit Tests
-Add unit tests under [sglang/test](https://github.com/sgl-project/sglang/tree/main/test). You can learn how to add and run tests from the README.md in that folder.
 
-## For Newcomers
-If you want to contribute or learn but don't have a concrete idea yet, you can pick a task labeled as "good first issue" or "help wanted" from the list below.
-[https://github.com/sgl-project/sglang/issues?q=is%3Aissue+label%3A%22good+first+issue%22%2C%22help+wanted%22](https://github.com/sgl-project/sglang/issues?q=is%3Aissue+label%3A%22good+first+issue%22%2C%22help+wanted%22)
+- **pre-commit install**: Installs the Git hook, which automatically runs style checks on staged files.
+- **pre-commit run --all-files**: Manually runs all the configured checks on your repository, applying automatic fixes where possible.
+
+Please ensure your code passes all these checks locally before creating a Pull Request.
+
+## 3. Writing Documentation & Running Docs CI
+
+All documentation files are located in the docs folder. We use make commands for building, cleaning, and previewing the documentation. Below is the typical workflow:
+
+1. **Compile Jupyter Notebooks**
+
+bash
+   make compile
+
+   This command processes all .ipynb files into intermediate formats.
+
+2. **Generate HTML Documentation**
+
+bash
+   make html
+
+   The generated static site can be found under the _build/html directory.
+
+3. **Local Preview**
+
+bash
+   bash serve.sh
+
+   Open your browser at the specified port to view the docs locally.
+
+4. **Clean Notebook Outputs & Build Artifacts**
+
+bash
+   find . -name '*.ipynb' -exec nbstripout {} \;
+   make clean
+
+   - **nbstripout** removes all outputs from Jupyter notebooks.
+   - **make clean** removes temporary files and _build artifacts.
+
+5. **Modify index.rst (If Necessary)**
+   Update index.rst whenever you add or remove sections from the documentation.
```

Comment:
**We prefer Jupyter Notebook over Markdown files since its correctness can be ensured by our docs CI.**

First make a new file under approperiate path and modify `index.rst` if neccessary.

```bash
# Compile all .ipynb and markdown files
# The jupyter notebooks will be executed

make compile

# Generate static files and they can be found under _build/html
make html

# Preview the docs
# Open your browser at the specified port to view the docs locally

bash serve.sh
```

Once you are sure your docs are right and passed compile, make a PR.

```bash
# Clean Notebook outputs to make your PR clear

pip install nbstripout
find . -name '*.ipynb' -exec nbstripout {} \;

# Run pre-commit

pre-commit run --all-files
```

Then you can submit your PR.

TODO: How to run server/engine and close server/engine in docs.